frontend/public/components/modals/roll-back-rc-modal.jsx

```diff
@@ -0,0 +1,107 @@
+import * as React from 'react';
+import { Trans, useTranslation } from 'react-i18next';
+import { createModalLauncher, ModalTitle, ModalBody, ModalSubmitFooter } from '../factory/modal';
+import { withHandlePromise } from '../utils';
+import { DeploymentConfigModel } from '../../models';
+import { k8sCreate, k8sUpdate } from '../../module/k8s';
+
+const BaseRollBackRCModal = withHandlePromise((props) => {
+  const { t } = useTranslation();
+  const dcName = props.resource.metadata.ownerReferences[0].name;
+  const [changeScaleSettings, setChangeScaleSettings] = React.useState(false);
+  const [changeStrategy, setChangeStrategy] = React.useState(false);
+  const [changeTriggers, setChangeTriggers] = React.useState(false);
+
+  const submit = (e) => {
+    e.preventDefault();
+
+    // put together a new rollback request
+    const req = {
+      kind: 'DeploymentConfigRollback',
+      apiVersion: 'apps.openshift.io/v1',
+      name: dcName,
+      latest: true,
+      force: true,
+      spec: {
+        from: {},
+        revision: parseInt(
+          props.resource?.metadata?.annotations?.['openshift.io/deployment-config.latest-version'],
+          10,
+        ),
+        includeTemplate: true,
+        includeReplicationMeta: changeScaleSettings,
+        includeStrategy: changeStrategy,
+        includeTriggers: changeTriggers,
+      },
+    };
+    const opts = {
+      name: dcName,
+      ns: props.resource.metadata.namespace,
+      path: 'rollback',
+    };
+
+    return props.handlePromise(
+      // create the deployment config rollback
+      k8sCreate(DeploymentConfigModel, req, opts).then((updatedDC) => {
+        // update the deployment config based on the one returned by the rollback
+        return k8sUpdate(DeploymentConfigModel, updatedDC);
+      }),
+      props.close,
+    );
+  };
+
+  return (
+    <form onSubmit={submit} name="form" className="modal-content">
+      <ModalTitle>{t('modal~Roll Back')}</ModalTitle>
+      <ModalBody>
+        <p>
+          <Trans t={t} ns="modal">
+            Use the following settings from{' '}
+            <strong className="co-break-word">
+              {{ resourceName: props.resource.metadata.name }}
+            </strong>{' '}
+            when rolling back:
+          </Trans>
+        </p>
+        <div className="checkbox">
+          <label className="control-label">
+            <input
+              type="checkbox"
+              onChange={() => setChangeScaleSettings(!changeScaleSettings)}
+              checked={!!changeScaleSettings}
+            />
+            {t('modal~replica count and selector')}
+          </label>
+        </div>
+        <div className="checkbox">
+          <label className="control-label">
+            <input
+              type="checkbox"
+              onChange={() => setChangeStrategy(!changeStrategy)}
+              checked={!!changeStrategy}
```

Comment:
These values should already be booleans, correct? I don't think the `!!` is necessary.